element_interface/intanloader/rhsutilities.py

```diff
@@ -0,0 +1,1037 @@
+                        # header['supply_voltage_channels'].append(new_channel)
+                    elif signal_type == 3:
+                        header["board_adc_channels"].append(new_channel)
+                    elif signal_type == 4:
+                        header["board_dac_channels"].append(new_channel)
+                    elif signal_type == 5:
+                        header["board_dig_in_channels"].append(new_channel)
+                    elif signal_type == 6:
+                        header["board_dig_out_channels"].append(new_channel)
+                    else:
+                        raise Exception("Unknown channel type.")
+
+    # Summarize contents of data file.
+    header["num_amplifier_channels"] = len(header["amplifier_channels"])
+    header["num_board_adc_channels"] = len(header["board_adc_channels"])
+    header["num_board_dac_channels"] = len(header["board_dac_channels"])
+    header["num_board_dig_in_channels"] = len(header["board_dig_in_channels"])
+    header["num_board_dig_out_channels"] = len(header["board_dig_out_channels"])
+
+    return header
+
+
+# Define notch_filter function
+def notch_filter(input, f_sample, f_notch, bandwidth):
+    """Implements a notch filter (e.g., for 50 or 60 Hz) on vector 'input'.
+
+    f_sample = sample rate of data (input Hz or Samples/sec)
+    f_notch = filter notch frequency (input Hz)
+    bandwidth = notch 3-dB bandwidth (input Hz).  A bandwidth of 10 Hz is
+    recommended for 50 or 60 Hz notch filters; narrower bandwidths lead to
+    poor time-domain properties with an extended ringing response to
+    transient disturbances.
+
+    Example:  If neural data was sampled at 30 kSamples/sec
+    and you wish to implement a 60 Hz notch filter:
+
+    output = notch_filter(input, 30000, 60, 10);
+    """
+    t_step = 1.0 / f_sample
+    f_c = f_notch * t_step
+
+    L = len(input)
+
+    # Calculate IIR filter parameters
+    d = math.exp(-2.0 * math.pi * (bandwidth / 2.0) * t_step)
+    b = (1.0 + d * d) * math.cos(2.0 * math.pi * f_c)
+    a0 = 1.0
+    a1 = -b
+    a2 = d * d
+    a = (1.0 + d * d) / 2.0
+    b0 = 1.0
+    b1 = -2.0 * math.cos(2.0 * math.pi * f_c)
+    b2 = 1.0
+
+    output = np.zeros(len(input))
+    output[0] = input[0]
+    output[1] = input[1]
+    # (If filtering a continuous data stream, change output[0:1] to the
+    #  previous final two values of output.)
+
+    # Run filter
+    for i in range(2, L):
+        output[i] = (
+            a * b2 * input[i - 2]
+            + a * b1 * input[i - 1]
+            + a * b0 * input[i]
+            - a2 * output[i - 2]
+            - a1 * output[i - 1]
+        ) / a0
+
+    return output
+
+
+# Define find_channel_in_group function
+def find_channel_in_group(channel_name, signal_group):
+    for count, this_channel in enumerate(signal_group):
+        if this_channel["custom_channel_name"] == channel_name:
+            return True, count
+    return False, 0
```

Comment:
By reserving `-1` as your not-found value, you could have this function only return the `this_channel` value and not need to juggle two returns

Reply:
It's not my code.